lcmark.lua

```diff
@@ -19,21 +18,34 @@ lcmark.writers = {
   commonmark = cmark.render_commonmark
 }
 
-local default_yaml_parser = function(...)
-  if not yaml then
-    local success, loaded = pcall(require, "yaml")
+local default_yaml_parser = nil
 
-    if success and type(loaded.load) == "function" then
-      yaml = loaded
-    else
-      error("Failed to load the 'yaml' library. Are you sure you have the " ..
-            "correct library installed and are passing the correct options?", 0)
-    end
+local function try_load(module_name, func_name)
+  if default_yaml_parser then -- already loaded; skip
+    return
+  end
+
+  local success, loaded = pcall(require, module_name)
+
+  if not success then
+    return
+  end
+  if type(loaded) ~= "table" or type(loaded[func_name]) ~= "function" then
+    return
   end
 
-  return yaml.load(...)
+  default_yaml_parser = loaded[func_name]
+  lcmark.yaml_parser_name = module_name .. "." .. func_name
 end
 
+try_load("lyaml", "load")
+try_load("yaml", "load") -- must come before yaml.eval
+try_load("yaml", "eval")
```

Comment:
Would it make this slightly faster if we did
```lua
try_load("lyaml","load") or try_load("yaml", "load") or try_load("yaml","eval")
```
?

Reply:
Yes, in theory, it would be approximately two function calls faster. However, I'm pretty sure the performance savings are negligible since this code runs only once (when the module is initially `require()`d).

Also, I'd like to keep the calls on separate lines, which IMO makes it easier to see which modules it's trying to load.

Reply:
Fair enough.